### PR TITLE
fs: Move Kconfig options to fs dedicated files

### PR DESCRIPTION
**Pull request by de-nordic**
This commit moves LittleFS related Kconfig options to Kconfig.littlefs and FATFS related options to Kconfig.fatfs.

### CODE REVIEW

File: subsys/fs/Kconfig.littlefs [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
 # SPDX-License-Identifier: Apache-2.0
 
+config FILE_SYSTEM_LITTLEFS
+	bool "LittleFS support"
+	depends on FLASH_MAP
+	depends on FLASH_PAGE_LAYOUT
+	help
+	  Enables LittleFS file system support.
+
 menu "LittleFS Settings"
```
Comment (galak) [inline subsys/fs/Kconfig.littlefs:13]: Wonder if this and other should be wrapped with `if FILE_SYSTEM_LITTLEFS` (similar to how various `drivers/serial/Kconfig.*` files are structured).

**Review by nashif — APPROVED** [review]

File: subsys/fs/Kconfig.littlefs (hunk 1 above)
Comment (de-nordic) [reply subsys/fs/Kconfig.littlefs:13]: > 
> 
> Wonder if this and other should be wrapped with `if FILE_SYSTEM_LITTLEFS` (similar to how various `drivers/serial/Kconfig.*` files are structured).

The main `Kconfig` for `fs` does so, but if we are trying to be consistent across various subsystems, I may do the `if` for both the `Kconfig.littlefs` and `Kconfig.fatfs`.
I can not find an example within `drivers/serial/` that does so, can you point me out to such file?

Comment (galak) [reply subsys/fs/Kconfig.littlefs:13]: `drivers/serial/Kconfig.native_posix` was the example I looked at.

File: subsys/fs/Kconfig.fatfs [hunk 2]
```diff
@@ -0,0 +1,98 @@
+# Copyright (c) 2016 Intel Corporation
+# Copyright (c) 2020 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+config FAT_FILESYSTEM_ELM
+	bool "ELM FAT file system support"
+	depends on FILE_SYSTEM
+	select DISK_ACCESS
+	help
+	  Use the ELM FAT File system implementation.
```
Comment (de-nordic) [inline subsys/fs/Kconfig.fatfs:10]: @galak Is the `depends on FILE_SYSTEM` what has been missing?

**Comment by galak** [comment]
So if you build something like `tests/subsys/fs/fat_fs_api` and look at the zephyr/.config you'll see something like:

```
CONFIG_FS_LITTLEFS_NUM_FILES=4
CONFIG_FS_LITTLEFS_NUM_DIRS=4
CONFIG_FS_LITTLEFS_READ_SIZE=16
CONFIG_FS_LITTLEFS_PROG_SIZE=16
CONFIG_FS_LITTLEFS_CACHE_SIZE=64
CONFIG_FS_LITTLEFS_LOOKAHEAD_SIZE=32
CONFIG_FS_LITTLEFS_BLOCK_CYCLES=512
```

So I think we need something like:

```
diff --git a/subsys/fs/Kconfig.littlefs b/subsys/fs/Kconfig.littlefs
index eac7637583..b1fb90ad81 100644
--- a/subsys/fs/Kconfig.littlefs
+++ b/subsys/fs/Kconfig.littlefs
@@ -11,6 +11,8 @@ config FILE_SYSTEM_LITTLEFS
        help
          Enables LittleFS file system support.
 
+if FILE_SYSTEM_LITTLEFS
+
 menu "LittleFS Settings"
        visible if FILE_SYSTEM_LITTLEFS
 
@@ -98,3 +100,5 @@ config FS_LITTLEFS_FC_MEM_POOL_NUM_BLOCKS
 endif # FS_LITTLEFS_FC_MEM_POOL
 
 endmenu
+
+endif # FILE_SYSTEM_LITTLEFS
```

For each of the filesystem's.

File: subsys/fs/Kconfig.fatfs (hunk 2 above)
Comment (galak) [reply subsys/fs/Kconfig.fatfs:10]: Sorry for delay, responded below with what I think is missing.

**Review by galak — APPROVED** [review]

**Pull request merged**